app/Http/Controllers/Auth/RegisteredUserController.php

```diff
@@ -24,16 +26,27 @@
      */
     public function create(): View
     {
-        return view('auth.register');
+        $githubEmail = session()->pull('github_email');
+        $githubUsername = session()->pull('github_username');
+
+        return view('auth.register')->with('githubEmail', $githubEmail)->with('githubUsername', $githubUsername);
     }
 
     /**
      * Handle an incoming registration request.
      *
      * @throws \Illuminate\Validation\ValidationException
      */
-    public function store(Request $request): RedirectResponse
+    public function store(Request $request, GitHub $gitHub): RedirectResponse
     {
+        $githubUsername = $request->github_username;
```

Comment:
If we receive a GitHub username, we save again these two fields in the session in case there is a validation error and the user is redirected back to the register page